spec/asyncapi.md

```diff
@@ -1882,6 +1874,37 @@ components:
             maximum: 100
 ```
 
+#### <a name="multiFormatSchemaObject"></a>Multi Format Schema Object
+
+The Multi Format Schema Object allows to define non AsyncAPI schema.
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---:|---
+<a name="schemaObjectSchemaFormat"></a>schemaFormat | `string` | **Required**. A string containing the name of the schema format used to define the message payload. If omitted, implementations should parse the payload as a [Schema object](#schemaObject). When the payload is defined using a `$ref` to a remote file, it is RECOMMENDED the schema format includes the file encoding type to allow implementations to parse the file correctly. E.g., adding `+yaml` if content type is `application/vnd.apache.avro` results in `application/vnd.apache.avro+yaml`.<br/><br/>Check out the [supported schema formats table](#multiFormatSchemaFormatTable) for more information. Custom values are allowed but their implementation is OPTIONAL. A custom value MUST NOT refer to one of the schema formats listed in the [table](#multiFormatSchemaFormatTable).<br/><br/>When using a [Reference Object](#referenceObject) within the schema, the `schemaFormat` of the reference target SHOULD be exactly the same as of the `schema` containing the reference.
```

Comment:
```suggestion
<a name="multiFormatSchemaObjectSchemaFormat"></a>schemaFormat | `string` | **Required**. A string containing the name of the schema format that is used to define the information. 
If `schemaFormat` is missing, it MUST default to `application/vnd.aai.asyncapi+json;version={{asyncapi}}` where `{{version}}` matches the [AsyncAPI Version String](#A2SVersionString). In such a case, this would make the Multiformat Schema Object equivalent to the [Schema Object](#schemaObject). When the schema is defined using a [Reference Object](#referenceObject) to a remote resource, it is RECOMMENDED the schema format includes the file encoding to help implementations parse the information correctly. E.g., adding `+yaml` if the content type is `application/vnd.apache.avro` results in `application/vnd.apache.avro+yaml`.<br/><br/>Check out the [supported schema formats table](#multiFormatSchemaFormatTable) for more information. Custom values are allowed but their implementation is OPTIONAL. A custom value MUST NOT refer to one of the schema formats listed in the [table](#multiFormatSchemaFormatTable).<br/><br/>When using [Reference Objects](#referenceObject) within the schema, the `schemaFormat` of the referenced resource MUST match the `schemaFormat` of the schema containing the reference.
```

> When using [Reference Objects](#referenceObject) within the schema, the `schemaFormat` of the referenced resource MUST match the `schemaFormat` of the schema containing the reference.

Would be cool if someone can help us rephrase this. It's really hard to read :sweat_smile:

Reply:
how about

> When using [Reference Objects](https://github.com/asyncapi/spec/pull/910/files#referenceObject) within the `schema`, the `schemaFormat` of the resource being referenced MUST match the `schemaFormat` of the `schema` that contains the initial reference. For example, if you reference Avro schema, then `schemaFormat` of referencing resource and the resource being reference MUST match.

?

Reply:
I used proposal of @derberg